src/Invictus.Testing/Model/LogicApp.cs

```diff
@@ -14,19 +14,5 @@ public class LogicApp
         public string Version { get; set; }
         public string AccessEndpoint { get; set; }
         public dynamic Definition { get; set; }
-
-        public static explicit operator LogicApp(Workflow workflow)
-        {
```

Comment:
Removed all this explicit operators so we have more control and less 'hiding' (just following Arcus guidliens). I'm wondering why we have separate models here? For separation and more control if the Azure packages change maybe?

Reply:
It would make it immutable and not use `dynamic` here on all types (in other PR of course).

Reply:
The problem is that the Azure models did, at the time of writing, not contain a full model for the actual logic app definition, but it was considered a string-object instead, containing the json-structured definition. This meant that extracting the actions to be updated (in case of mocking) was practically impossible.

Going for `dynamic` type was an explicit choice, since for example for the trackedProperties-object (and others), this can be defined as a string-array in the logic app definition, but you also have to possibility to directly link the output, for example:

```json
"trackedProperties": {
     "Status": "Processing",
     "DocumentNumber": "1234"
} 
```
But the following is also possible, as this is referring to the context-output of a different logic app action:
```json
"trackedProperties": "@body('Matrix_-_Promote_Tracking_Properties')?['Context']",
```

This means not all actions and/or properties always have a fixed structure. Using dynamic allows the framework to correctly process them and allows the client using the framework to correctly access the underlying structure.

_If anything about this is unclear, let me know!_

Reply:
Ok, but I can still remove the explicit operations here since this makes mapping bit hidden for devs. Also, is there a reason we have separate models? Maybe to not depend on the Azure SDK in our testing library?

For the dynamic property, I figured it had to be something like that as you explained. Hmm, I wonder if we could classify the different type of structures and make use of the newtonsoft converters. But that could maybe overcomplicate things.

Reply:
The explicit operators can be removed indeed.

I think the main reason why the Azure SDK wasn't used was to reduce the amount of dependencies and thus remove an additional "risk" that could perhaps break the framework, but if I'm not mistaken some properties were still missing from the models 
Looking back at this I don't see any reason not to refer to the Azure SDK, since if those models would change, it means the actual Logic App structures would be modified as well and our framework would be broken any way.

Before converting to make use of the Azure SDK, perhaps we should have a look at the current models in the SDK and check whether these definitely contain any/all properties required from the framework point of view.

Regarding the newtonsoft converters: that would definitely be an option, but at this point I would stick with the `dynamic` types and lets perhaps come back to this, once everything else is finished or if we run into an issue due to this of course.

Reply:
>Regarding the newtonsoft converters: that would definitely be an option, but at this point I would stick with the dynamic types and lets perhaps come back to this, once everything else is finished or if we run into an issue due to this of course.

Yes, of course! I'll add an issue for this so we dont forget.

Reply:
> Before converting to make use of the Azure SDK, perhaps we should have a look at the current models in the SDK and check whether these definitely contain any/all properties required from the framework point of view.

The SDK definitly contains additional properties that aren't defined in our model. In terms of 'required' properties, that's maybe a question for users of the framework.

Reply:
> The explicit operators can be removed indeed.

![](https://media.giphy.com/media/DffShiJ47fPqM/source.gif)

Reply:
> > The explicit operators can be removed indeed.
> 
> ![](https://camo.githubusercontent.com/eb44d9508e1e76cb6e7bd9b32982ec979b213110/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f4466665368694a34376650714d2f736f757263652e676966)

😂

Reply:
Seems like the `LogicAppRun` model is also containing the actions on the workflow. So it's actually two SDK models in one.

Reply:
Added #35 for this.